devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy

```diff
@@ -198,13 +199,26 @@ private Map getContactResult(value) {
 }
 
 private Map getAnalogInputResult(value) {
-    Float f = Float.intBitsToFloat(value.intValue())
-    int pc = f.round(0)
-    String descriptionText = "${device.displayName} : $pc" 
+    Float fpc = Float.intBitsToFloat(value.intValue())
+    def prevInOut = device.currentState('inOutDir')?.value
+    int pc = ((int)(fpc*10))/10 //people counter
+    int inout = ((int)(fpc*10).round(0))%10; // inout direction : .1 = in, .2 = out, .0 = ready
+    if(inout>2) inout = 2
+    String inoutString = ( (inout==1) ? "in" : (inout==2) ? "out":"ready")
+    String descriptionText1 = "${device.displayName} : $pc"
+    String descriptionText2 = "${device.displayName} : $inoutString"
+    log.debug "[$fpc] = people: $pc, dir: $inout, $inoutString"
+    
+    if((inout != "ready") && (prevInOut == inoutString)) {
+        sendEvent(name: "inOutDir", value: "ready", displayed: true)
+    }
```

Comment:
double checking -> 
Since this function already returns `inOutDir` event, why is this necessary to send this `inOutDir (ready)` event here?

Reply:
for continuous in or out event.